### PR TITLE
[react-events] DOM event testing library

### PR DESCRIPTION
**Pull request by necolas**
This patch formalizes the mock native events and event sequences used in unit tests.

The `createEventTarget` function returns an object that can be used to dispatch native event sequences on the target without having to manually do so across all the scenarios we need to account for. Unit tests can be written as if we were only working with PointerEvent, but they will dispatch realistic native event sequences based on the execution environment (e.g., is PointerEvent supported?) and pointer type.

```
describe.each(environments)('Suite', (hasPointerEvents) => {
  beforeEach(() => {
    // setup
  });

  test.each(pointerTypes)('Test', (pointerType) => {
    const target = createEventTarget(node);
    target.pointerdown({pointerType});
    expect(callback).toBeCalled();
  });
});
```

Every native event that is dispatched now includes a complete object by default. The properties of the events can be customized. Properties that shouldn't be relied on in responder implementations are excluded from the mock native events to ensure tests will fail. Equivalent properties are normalized across different event types, e.g., 'pointerId' is converted to 'identifier' before a TouchEvent is dispatched.

### CODE REVIEW

File: packages/react-events/src/dom/__tests__/Press-test.internal.js [hunk 1]
```diff
@@ -140,51 +127,54 @@ describe.each(environmentTable)('Press responder', hasPointerEvents => {
       );
     });
 
-    it('is not called after "pointermove" following auxillary-button press', () => {
-      const target = ref.current;
-      target.getBoundingClientRect = () => ({
+    it('is not called after pointer move following auxillary-button press', () => {
+      const node = ref.current;
+      const target = createEventTarget(node);
+      node.getBoundingClientRect = () => ({
```
Comment (necolas) [inline packages/react-events/src/dom/__tests__/Press-test.internal.js]: Use `setBoundingClientRect`

File: packages/react-events/src/dom/__tests__/Press-test.internal.js [hunk 2]
```diff
@@ -475,20 +463,21 @@ describe.each(environmentTable)('Press responder', hasPointerEvents => {
     it.each(pointerTypesTable)(
       'is called after pointer move: %s',
       pointerType => {
-        const target = ref.current;
-        target.getBoundingClientRect = () => ({
+        const node = ref.current;
+        node.getBoundingClientRect = () => ({
```
Comment (necolas) [inline packages/react-events/src/dom/__tests__/Press-test.internal.js]: use target helper

File: packages/react-events/src/dom/testing-library/domEventSequences.js [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+    if (hasPointerEvent()) {
+      dispatch(domEvents.pointerup(payload));
+      dispatch(domEvents.lostpointercapture(payload));
+      dispatch(domEvents.pointerout(payload));
+      dispatch(domEvents.pointerleave(payload));
+    }
+    dispatch(
+      domEvents.touchend(
+        payload,
+        payload != null ? payload.pointerId : undefined,
```
Comment (necolas) [inline packages/react-events/src/dom/testing-library/domEventSequences.js]: Remove

File: packages/react-events/src/dom/testing-library/domEventSequences.js [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+  } else {
+    if (hasPointerEvent()) {
+      dispatch(domEvents.pointerover(payload));
+      dispatch(domEvents.pointerenter(payload));
+      dispatch(domEvents.pointerdown(payload));
+    }
+    dispatch(
+      domEvents.touchstart(
+        payload,
+        payload != null ? payload.pointerId : undefined,
```
Comment (necolas) [inline packages/react-events/src/dom/testing-library/domEventSequences.js]: Remove

File: packages/react-events/src/dom/testing-library/domEventSequences.js [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+  if (hasPointerEvent()) {
+    dispatch(domEvents.pointermove(payload));
+  }
+  if (pointerType === 'mouse') {
+    dispatch(domEvents.mousemove(payload));
+  } else {
+    dispatch(
+      domEvents.touchmove(
+        payload,
+        payload != null ? payload.pointerId : undefined,
```
Comment (necolas) [inline packages/react-events/src/dom/testing-library/domEventSequences.js]: Remove

File: packages/react-events/src/dom/testing-library/index.js [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        left: x,
+        right: x + width,
+        top: y,
+        bottom: y + height,
+      };
+    };
+  },
+});
+
+module.exports = {
```
Comment (trueadm) [inline packages/react-events/src/dom/testing-library/index.js]: Can't we just use ES module exports here? It's only going to be used in Jest, but it makes things more consistent with the other files in `src`.

Comment (necolas) [reply packages/react-events/src/dom/testing-library/index.js]: Sure. Spent too much time in www :P

**Pull request merged**